modules/base/rendering/renderabletrail.cpp

```diff
@@ -23,346 +23,355 @@
+        if (renderLines) {
+            p->setUniform("renderPhase", RenderPhaseLines);
+            // Subclasses of this renderer might be using the index array or might now be
+            // so we check if there is data available and if there isn't, we use the
+            // glDrawArrays draw call; otherwise the glDrawElements
+            if (info._iBufferID == 0) {
+                glDrawArrays(
+                    GL_LINE_STRIP,
+                    info.first,
+                    info.count
+                );
+            }
+            else {
+                glDrawElements(
+                    GL_LINE_STRIP,
+                    info.count,
+                    GL_UNSIGNED_INT,
+                    reinterpret_cast<void*>(info.first * sizeof(unsigned int))
+                );
+            }
         }
-
-        if (!failed || intervalSet) {
-            try {
-                p =
-                    SpiceManager::ref().targetPosition(_target, _observer, _frame, {}, time, lightTime);
+        if (renderPoints) {
+            // Subclasses of this renderer might be using the index array or might now be
+            // so we check if there is data available and if there isn't, we use the
+            // glDrawArrays draw call; otherwise the glDrawElements
+            p->setUniform("renderPhase", RenderPhasePoints);
+            if (info._iBufferID == 0) {
+                glDrawArrays(GL_POINTS, info.first, info.count);
             }
-            catch (const SpiceManager::SpiceException& e) {
-                // This fires for PLUTO BARYCENTER and SUN and uses the only value sometimes?
-                // ---abock
-                //LERROR(e.what());
-                failed = true;
+            else {
+                glDrawElements(
+                    GL_POINTS,
+                    info.count,
+                    GL_UNSIGNED_INT,
+                    reinterpret_cast<void*>(info.first * sizeof(unsigned int))
+                );
             }
         }
-        
-        psc pscPos = PowerScaledCoordinate::CreatePowerScaledCoordinate(p.x, p.y, p.z);
-        pscPos[3] += 3;
-
-        _vertexArray[i] = {pscPos[0], pscPos[1], pscPos[2], pscPos[3]};
-        time -= _increment;
+    };
+
+    // The combined size of vertices; -1 because we duplicate the penultimate point
+    int totalNumber =
+        _primaryRenderInformation.count + _floatingRenderInformation.count - 1;
+
+    // The primary information might use an index buffer, so we might need to start at an
+    // offset
+    int primaryOffset =
+        _primaryRenderInformation._iBufferID == 0 ? 0 : _primaryRenderInformation.first;
+
+    // Render the primary batch of vertices
+    render(_primaryRenderInformation, totalNumber, primaryOffset);
+
+    // The secondary batch is optional,. so we need to check whether we have any data here
+    if (_floatingRenderInformation._vaoID == 0 || _floatingRenderInformation.count == 0) {
+        render(
```

Comment:
It was actually used twice.. in the line you highlight and just a line before then